### PR TITLE
Add glacier_lengths

### PR DESCRIPTION
**Pull request by erikmannerfelt**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by erikmannerfelt** [comment]
Hey, I'd love to know what is going wrong here! I've tried it multiple times on my machine (Arch Linux 64bit), and it works well, but it seems like the Azure builds can't import the package after successfully building.

```bash
❯ conda info                                                                                                                                         16:13

     active environment : swissterra
    active env location : /home/erik/.local/share/conda/swissterra
            shell level : 3
       user config file : /home/erik/.condarc
 populated config files : /home/erik/.local/miniforge3/.condarc
                          /home/erik/.condarc
          conda version : 4.9.2
    conda-build version : not installed
         python version : 3.9.0.final.0
       virtual packages : __cuda=11.2=0
                          __glibc=2.33=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/erik/.local/miniforge3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/erik/.cache/conda
       envs directories : /home/erik/.local/share/conda
                          /home/erik/.local/miniforge3/envs
                          /home/erik/.conda/envs
               platform : linux-64
             user-agent : conda/4.9.2 requests/2.25.1 CPython/3.9.0 Linux/5.11.11-arch1-1 arch/rolling glibc/2.33
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

❯ conda-build -V                                                                                                                                     16:13
conda-build 3.21.4

File: recipes/glacier_lengths/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{% set name = 'glacier_lengths' %}
+{% set version = '0.0.4' %}
+
+package:
+  name: '{{ name | lower }}'
+  version: '{{ version }}'
+
+source:
+  url: https://github.com/erikmannerfelt/glacier_lengths/archive/refs/tags/v0.0.4.tar.gz
```
Comment (wolfv) [inline recipes/glacier_lengths/meta.yaml:9]: can you use `https://github.com/erikmannerfelt/glacier_lengths/archive/refs/tags/v{{ version }}.tar.gz`? That way automatic updates from the bots will work fine.

Comment (erikmannerfelt) [reply recipes/glacier_lengths/meta.yaml]: Changed in 2469ca2!

File: recipes/glacier_lengths/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  number: 0
+  noarch: python
+  script: '{{ PYTHON }} -m pip install . -vv'
+
+requirements:
+  host:
+  - python >=3.7
+  - pip
+  - shapely
+  - numpy
```
Comment (ocefpaf) [inline recipes/glacier_lengths/meta.yaml:22]: You probably don't need shapely and numpy on host, unless this package imports itself at setup.py.

Comment (erikmannerfelt) [reply recipes/glacier_lengths/meta.yaml:22]: The package is imported to fetch the version. Maybe it should be the other way haha.. Shapely and numpy are the only dependencies, however, so it's pretty fast, right?

Comment (ocefpaf) [reply recipes/glacier_lengths/meta.yaml:22]: No other way unless they fix it upstream :-(

**Pull request merged**